packages/rest/src/writer.ts

```diff
@@ -30,6 +31,16 @@ export function writeResultToResponse(
     return;
   }
 
+  // result is of type RedirectRoute
+  const isRedirect = result instanceof RedirectRoute;
+  // Controller methods can return a redirect handler to handle redirections.
+  // This will be useful in certain oauth flows like `authorization code`
+  // , where the authenticating endpoint needs to redirect to a third-party oauth provider
+  if (isRedirect) {
```

Comment:
We don't have to handle `RedirectRoute` operation result in the `writer.ts`.  The controller can handle redirect and simple return `response` as is to skip response serialization.

Reply:
agreed

Reply:
@raymondfeng on second thought, I think having the controller return a simple redirect object makes the design very neat and clear.

Reply:
this would also make user code more unit testable.

Reply:
I'm not a big fan of hard-coding a lot of flavors in the writer unless we introduce extensibility of response handlers. Meanwhile, I won't block the PR either.

Reply:
@raymondfeng to an extent I do agree with your view. I will make the change.
I am also going to see what other reviewers comment.

Reply:
We should remove this condition now as `redirect` is already handled in the controller.